### PR TITLE
Add a scaling parameter for all element integral postprocessors

### PR DESCRIPTION
**Pull request by YaqiWang**
I am sending this PR because I found it is quite complicated to add a scaling factor for a postprocessor, similar like the issue I reported in  #5954. This PR will be a quick solution for me although it can be solved by #5954. I still think this can be useful by itself. Merge it or not as your wish. Thanks.

### CODE REVIEW

**Comment by permcody** [comment]
I'm fine with this, but this is the kind of little change that sometimes gets @friedmud going though... Maybe some of the other @idaholab/moose-developers have some thoughts on this?

File: framework/src/postprocessors/ElementW1pError.C [hunk 1]
```diff
@@ -34,7 +34,8 @@ ElementW1pError::ElementW1pError(const InputParameters & parameters) :
 Real
 ElementW1pError::getValue()
 {
-  return std::pow(ElementIntegralPostprocessor::getValue(), 1./_p);
```
Comment (dschwen) [inline framework/src/postprocessors/ElementW1pError.C]: Why did you get rid of all the calls to the base class?

Comment (permcody) [inline framework/src/postprocessors/ElementW1pError.C]: BASE CLASS!

@dschwen's correct, while your new statements might be equivalent now, if we change the BASE CLASS, things won't flow downstream anymore.

Comment (YaqiWang) [inline framework/src/postprocessors/ElementW1pError.C]: Because the base class apply the scaling factor too early.

Comment (permcody) [inline framework/src/postprocessors/ElementW1pError.C]: The right way would be to make then save the return value of that call to a local variable, then apply the scaling factor as you are doing before you return. The call should still be in there.

Comment (dschwen) [inline framework/src/postprocessors/ElementW1pError.C]: Right... Then the loss of elegance is outweighing the convenience of a scaling parameter. I'd vote against this change (if I had a vote :-D). Just use a second PP to scale the value.

**Comment by andrsd** [comment]
If we want to scale PPS, why not to add that parameter to `Postprocessor` class and do the scaling inside MOOSE, i.e. where the computed value gets stored. Similarly to what variable scaling does.  Just a thought...
BTW: I do not like this PR either ;-)

**Comment by andrsd** [comment]
Also, adding scaling to just element PPS is quarter-assed solution anyway ;-)

**Comment by YaqiWang** [comment]
To answer @dschwen, we do not have that scaling postprocessor yet... We could create a `ScalePostprocessor` then users could want a `BiasPostpostprocessor`... That is why I think #5954 should be the answer. Btw, `GatherSum, _integral_value` are both from base class anyway. We could create a base function `Real GatherAndReturn()` for this.

**Comment by YaqiWang** [comment]
@andrsd Scaling the postprocessor values sounds scary to me. I thought putting this parameter in Postprocessor but have to give it up because there will be lots more places to change due the the issue we see here...

**Comment by YaqiWang** [comment]
Or create a base function `Real getValueBeforeScale()`.

**Comment by dschwen** [comment]
Well, adding a `ScalePostprocessor` would almost be a smaller (certainly less invasive) change than this ;-). We Already have `DifferencePostprocessor` in the same spirit and we also have `FunctionValuePostprocessor` (which would probably make it necessary to add a Function object though, as I don't think you can use postprocessor names in the inline function definition).

**Comment by dschwen** [comment]
Well, adding a `ScalePostprocessor` would almost be a smaller (certainly less invasive) change than this ;-). We Already have `DifferencePostprocessor` in the same spirit and we also have `FunctionValuePostprocessor` (which would probably make it necessary to add a Function object though, as I don't think you can use postprocessor names in the inline function definition).

**Comment by friedmud** [comment]
I'm against this change.

Compute the Postprocessor like normal and then scale the value _outside_ the Postprocessor if you need it scaled.  This looks all simple... but what if you're scaling factor isn't constant?  What if you want to scale by the value of another Postprocessor?  Why are we only applying it to _these_ Postprocessors?

@dschwen is on the right track with a `ScalePostprocessor` or you can already do this using `FunctionValuePostprocessor` and dialing up any arbitrary operation involving Postprocessor values.

Don't take simple things and make them more complicated.  Use the simple things to build something more complicated.

**Comment by andrsd** [comment]
> Use the simple things to build something more complicated.

:+1:

**Comment by permcody** [comment]
Looks like we aren't going to merge this based on the comments. Can you propose a new PR following some of the comments here?

**Comment by YaqiWang** [comment]
Yes, I will add a `ScalePostprocessor`. I do not quite like this PR either. We could apply a function on the returned value of the current `getValue()`, but that will be too many changes and may not worthwhile.

**Pull request closed without merging**